hyperspy/io_plugins/tvips.py

```diff
@@ -0,0 +1,588 @@
+    nav_units = signal.axes_manager[0].units
+    nav_increment = signal.axes_manager[0].scale
+    try:
+        time_increment = (nav_increment * _ureg(nav_units)).to("ms").magnitude
+    except (AttributeError, pint.UndefinedUnitError, pint.DimensionalityError):
+        time_increment = 1
+    # imaging or diffraction
+    mode = kwds.pop("mode", None)
+    if mode is None:
+        mode = _guess_image_mode(signal)
+    mode = 2 if mode is None else mode
+    stagex = signal.metadata.get_item("Acquisition_instrument.TEM.Stage.x", 0)
+    stagey = signal.metadata.get_item("Acquisition_instrument.TEM.Stage.y", 0)
+    stagez = signal.metadata.get_item("Acquisition_instrument.TEM.Stage.z", 0)
+    stagea = signal.metadata.get_item("Acquisition_instrument.TEM.tilt_alpha", 0)
+    stageb = signal.metadata.get_item("Acquisition_instrument.TEM.tilt_beta", 0)
+    # TODO: is fcurrent actually beam current??
+    fcurrent = signal.metadata.get_item("Acquisition_instrument.TEM.beam_current", 0)
+    frames_to_save = num_frames
+    current_frame = 0
+    file_index = 0
+    with signal.unfolded(unfold_navigation=True, unfold_signal=False):
+        while (frames_to_save != 0):
+            suffix = "_" + (f"{file_index}".zfill(3))
+            filename = fnb + suffix + ext
+            if file_index == 0:
+                with open(filename, "wb") as f:
+                    main_header.tofile(f)
+                    file_location = f.tell()
+                    open_mode = "r+"
+            else:
+                file_location = 0
+                open_mode = "w+"
+            frames_saved = (max_file_size - file_location) // record_dtype.itemsize
+            # last file can contain fewer images
+            if frames_to_save < frames_saved:
+                frames_saved = frames_to_save
+            file_memmap = np.memmap(
+                filename, dtype=record_dtype, mode=open_mode,
+                offset=file_location, shape=frames_saved,
+            )
+            # fill in the metadata
+            file_memmap["mode"] = mode
+            file_memmap["stagex"] = stagex
+            file_memmap["stagey"] = stagey
+            file_memmap["stagez"] = stagez
+            file_memmap["stagea"] = stagea
+            file_memmap["stageb"] = stageb
+            file_memmap['fcurrent'] = fcurrent
+            rotator = np.arange(current_frame, current_frame + frames_saved)
+            milliseconds = rotator * time_increment
+            timestamps = (timestamp + milliseconds / 1000).astype(int)
+            milliseconds = milliseconds % 1000
+            file_memmap["timestamp"] = timestamps
+            file_memmap["ms"] = milliseconds
+            file_memmap["rotidx"] = rotator + 1
+            data = signal.data[current_frame:current_frame + frames_saved]
+            if signal._lazy:
+                if show_progressbar is None:
+                    show_progressbar = preferences.General.show_progressbar
```

Comment:
```suggestion
                show_progressbar = kwds.get(
                    'show_progressbar',
                    preferences.General.show_progressbar
                    )
```
And remove the line setting `show_progressbar` above

Reply:
done